### PR TITLE
feat(trends): Set default trend parameter based on project type

### PR DESCRIPTION
**Pull request by udameli**
When no trend parameter is set in query, get appropriate trend parameter based on the project type.

Fixes VIS-1275

### CODE REVIEW

File: static/app/views/performance/trends/utils.tsx [hunk 1]
```diff
@@ -162,6 +164,20 @@ export function getCurrentTrendParameter(location: Location): TrendParameter {
   return trendParameter || TRENDS_PARAMETERS[0];
 }
 
+export function performanceTypeToTrendParameterLabel(
+  performanceType: PROJECT_PERFORMANCE_TYPE
+): string {
+  switch (performanceType) {
+    case PROJECT_PERFORMANCE_TYPE.FRONTEND:
+    case PROJECT_PERFORMANCE_TYPE.FRONTEND_OTHER:
```
Comment (k-fish) [inline static/app/views/performance/trends/utils.tsx:172]: `FRONTEND_OTHER` is duration.

**Review by k-fish — COMMENTED** [review]
- Needs a test to check default param is set (trends/index.spec.jsx)
- You should check to see if this is making 2 api requests (1 actual / 1 cancelled) request since it's a didmount? In general I wouldn't fill in the url with the default, especially since pushing to browser history causes re-renders, I'd just have it use `param.query.trendsParameter || default`.

File: static/app/utils/performance/trends/trendsDiscoverQuery.tsx [hunk 2]
```diff
@@ -47,10 +50,14 @@ type EventProps = RequestProps & {
 };
 
 export function getTrendsRequestPayload(props: RequestProps) {
-  const {eventView} = props;
+  const {eventView, projects} = props;
```
[the remark this answers is not in the document]
Comment (udameli) [reply static/app/utils/performance/trends/trendsDiscoverQuery.tsx:53]: oh true, `useProjects` hook works better here 👍

[the remark this answers is not in the document]
Comment (udameli) [reply static/app/utils/performance/trends/trendsDiscoverQuery.tsx:53]: hm actually I'm going to revert and use the HOC here. `_GenericDiscoverQuery` is a class component (and a child of `GenericDiscoverQuery`) and it has quite a bit of logic in its lifecycle methods so I'm inclined to keep it as it is for now

[the remark this answers is not in the document]
Comment (dashed) [reply static/app/utils/performance/trends/trendsDiscoverQuery.tsx:53]: Ok sounds good.

File: tests/js/spec/views/performance/trends/utils.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  describe('getCurrentTrendParameter', function () {
+    it('returns trend parameter from location', () => {
+      const location = TestStubs.location({query: {trendParameter: 'FCP'}});
+      const expectedTrendParameter = {
+        label: 'FCP',
+        column: TrendColumnField.FCP,
+      };
+      // project with performance type 'any'
+      const projects = [TestStubs.Project({id: 1, plarform: null})];
```
Comment (dashed) [inline tests/js/spec/views/performance/trends/utils.spec.tsx:56]:
```suggestion
      const projects = [TestStubs.Project({id: 1, platform: null})];
```

File: tests/js/spec/views/performance/trends/utils.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    });
+
+    it('returns default trend parameter based on project type if no trend parameter set in location', function () {
+      const location = TestStubs.location();
+      const expectedTrendParameter = {
+        label: 'Duration',
+        column: TrendColumnField.DURATION,
+      };
+      // project with performance type 'any'
+      const projects = [TestStubs.Project({id: 1, plarform: null})];
```
Comment (dashed) [inline tests/js/spec/views/performance/trends/utils.spec.tsx:69]:
```suggestion
      const projects = [TestStubs.Project({id: 1, platform: null})];
```

Comment (udameli) [reply tests/js/spec/views/performance/trends/utils.spec.tsx:69]: 😶‍🌫️

File: static/app/views/performance/trends/index.tsx [hunk 5]
```diff
@@ -66,13 +66,14 @@ class TrendsSummary extends Component<Props, State> {
...
   renderContent() {
-    const {organization, location} = this.props;
+    const {organization, location, projects} = this.props;
     const {eventView} = this.state;
     return (
       <TrendsContent
         organization={organization}
         location={location}
         eventView={eventView}
+        projects={projects}
```
Comment (dashed) [inline static/app/views/performance/trends/index.tsx:76]: @udameli would you be able to add a test in https://github.com/getsentry/sentry/blob/a0cac1906303a5e097c479dde03120f46d9ff83b/tests/js/spec/views/performance/trends/index.spec.jsx ?

Comment (dashed) [reply static/app/views/performance/trends/index.tsx:76]: thanks! 🙏

**Review by dashed — APPROVED** [review]

**Pull request merged**